core/test/qa_Messages.cpp

```diff
@@ -18,509 +20,633 @@ auto boost::ut::cfg<boost::ut::override> = boost::ut::runner<boost::ut::reporter
-        sink.messageProcessor = messageProcessorCounter<"MulticastMessagingWithTheWorld_sink.messageProcessor">(sinkMessagesCountdown, collectedEventCount, "custom_kind"s, [] {
-            gr::Message outMessage;
-            outMessage[gr::message::key::Kind] = "custom_reply_kind";
-            return outMessage;
-        }());
-
-        expect(eq(ConnectionResult::SUCCESS, flow.connect<"out">(source).to<"in">(process)));
-        expect(eq(ConnectionResult::SUCCESS, flow.connect<"out">(process).to<"in">(sink)));
+const boost::ut::suite MessagesTests = [] {
+    using namespace std::string_literals;
+    using namespace boost::ut;
+    using namespace gr;
 
-        return Scheduler(std::move(flow));
-    }();
-
-    gr::testing::MessageSender<float> messageSender;
-    messageSender.messageGenerator = [&](auto * /*this*/) -> std::optional<gr::Message> {
-        if (!isDone()) {
-            gr::Message message;
-            message[gr::message::key::Kind]   = "custom_kind";
-            message[gr::message::key::Target] = "";
-            return message;
-        } else {
-            return {};
-        }
-    };
-    expect(eq(ConnectionResult::SUCCESS, messageSender.msgOut.connect(scheduler.msgIn)));
-
-    gr::testing::FunctionSink<float> messageReceiver;
-    messageReceiver.messageProcessor = [&](auto * /*_this*/, gr::MsgPortInNamed<"__Builtin"> & /*port*/, std::span<const gr::Message> messages) {
-        for (const auto &message : messages) {
-            const auto kind = gr::messageField<std::string>(message, gr::message::key::Kind);
-            if (kind == "custom_reply_kind") {
-                receiverGotAMessage = true;
-            }
-        }
+    static auto returnReplyMsg = [](gr::MsgPortIn &port) {
+        ConsumableSpan auto span = port.streamReader().get<SpanReleasePolicy::ProcessAll>(1UZ);
+        Message             msg  = span[0];
+        expect(span.consume(span.size()));
+        return msg;
     };
-    expect(eq(ConnectionResult::SUCCESS, scheduler.msgOut.connect(messageReceiver.msgIn)));
 
-    std::thread messenger([&] {
-        while (!isDone()) {
-            std::this_thread::sleep_for(100ms);
-            messageSender.processOne();
-            messageReceiver.processScheduledMessages();
-        }
-    });
-
-    scheduler.runAndWait();
-    messenger.join();
-}
+    "Block<T>-level message tests"_test = [] {
+        using namespace gr::testing;
+        using enum gr::message::Command;
+
+        "Block<T>-level heartbeat tests"_test = [] {
+            gr::MsgPortOut toBlock;
+            TestBlock<int> unitTestBlock({ { "name", "UnitTestBlock" } });
+            gr::MsgPortIn  fromBlock;
+
+            expect(eq(ConnectionResult::SUCCESS, toBlock.connect(unitTestBlock.msgIn)));
+            expect(eq(ConnectionResult::SUCCESS, unitTestBlock.msgOut.connect(fromBlock)));
+
+            "w/o explicit serviceName"_test = [&] {
```

Comment:
Would be good to add some documentation, that serviceName should be `""`, `Block::unique_name`, or `Block::name` or `unmatchedPropertyHandler` should be implemented.